Fixes time filter component

cqliu1 opened the pull request:
Closes #26639.

Time filter wasn't working correctly when setting the column to anything other than the default `@timestamp`. The cause was the time filter component was storing the initial value of the filter prop in its state, but that state wasn't getting updated when the prop value changed, which resulted in the `doSetFilter` handler always setting a filter with `@timestamp` as the column. I removed state in the time filter component, and now the component re-renders when the column changes.

elasticmachine commented:
Pinging @elastic/kibana-canvas

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4679/)

w33ble commented:
Nice find, clobbering prop values with state is no good. But this was added so that the element would update correctly, which it no longer does here:

Here, I pick some custom time range and apply it, and you can see the data is updated, but the element does not change. And when I open it again, it still says the filter is for the last 24 hours.

![dec-28-2018 11-50-36](https://user-images.githubusercontent.com/404731/50525154-f4cda200-0a96-11e9-9da3-a9bc9d2e3035.gif)

None of the props passed to `TimeFilter` change, and now that none of the state changes either, the element is never re-rendered to show the updated value.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/5289/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/5524/)

cqliu1 commented:
@w33ble I added state back to the time filter component, and now both the button and popover should be rendering with the correct values as filters change.

I also added min and max dates to the date picker, so you can't pick an end date earlier than your start date and a start date later than your end date.
